src/fund/components/FundCard.tsx

```diff
@@ -0,0 +1,164 @@
+import { useFundCardFundingUrl } from '../hooks/useFundCardFundingUrl';
+import { useFundCardSetupOnrampEventListeners } from '../hooks/useFundCardSetupOnrampEventListeners';
+import type { FundCardContentPropsReact, FundCardPropsReact } from '../types';
+import { FundButton } from './FundButton';
+import FundCardAmountInput from './FundCardAmountInput';
+import FundCardAmountInputTypeSwitch from './FundCardAmountInputTypeSwitch';
+import { FundCardHeader } from './FundCardHeader';
+import { FundCardPaymentMethodDropdown } from './FundCardPaymentMethodDropdown';
+import { FundCardProvider } from './FundCardProvider';
+import { useFundContext } from './FundCardProvider';
+
+export function FundCard({
+  children,
+  assetSymbol,
+  buttonText = 'Buy',
+  headerText,
+  paymentMethods = DEFAULT_PAYMENT_METHODS,
+}: FundCardPropsReact) {
+  const componentTheme = useTheme();
+
+  const {
+    amountInputComponent,
+    headerComponent,
+    amountInputTypeSwithComponent,
+    paymentMethodDropdownComponent,
+    submitButtonComponent,
+  } = useMemo(() => {
+    const childrenArray = Children.toArray(children);
+
+    return {
+      amountInputComponent: childrenArray.find(
+        findComponent(FundCardAmountInput),
+      ),
+      headerComponent: childrenArray.find(findComponent(FundCardHeader)),
+      amountInputTypeSwithComponent: childrenArray.find(
+        findComponent(FundCardAmountInputTypeSwitch),
+      ),
+      paymentMethodDropdownComponent: childrenArray.find(
+        findComponent(FundCardPaymentMethodDropdown),
+      ),
+      submitButtonComponent: childrenArray.find(findComponent(FundButton)),
+    };
+  }, [children]);
+
+  return (
+    <FundCardProvider asset={assetSymbol}>
+      <div
+        className={cn(
+          componentTheme,
+          background.default,
+          color.foreground,
+          'flex w-[440px] flex-col p-6',
+          text.headline,
+          border.radius,
+          border.lineDefault,
+        )}
+      >
+        <FundCardContent
+          assetSymbol={assetSymbol}
+          buttonText={buttonText}
+          headerText={headerText}
+          amountInputComponent={amountInputComponent}
+          headerComponent={headerComponent}
+          amountInputTypeSwithComponent={amountInputTypeSwithComponent}
+          paymentMethodDropdownComponent={paymentMethodDropdownComponent}
+          paymentMethods={paymentMethods}
+          submitButtonComponent={submitButtonComponent}
+        />
+      </div>
+    </FundCardProvider>
+  );
+}
+
+function FundCardContent({
+  assetSymbol,
+  buttonText = 'Buy',
+  headerText,
+  amountInputComponent,
+  headerComponent,
+  amountInputTypeSwithComponent,
+  paymentMethodDropdownComponent,
+  paymentMethods = DEFAULT_PAYMENT_METHODS,
+  submitButtonComponent,
+}: FundCardContentPropsReact) {
+  /**
+   * Fetches and sets the exchange rate for the asset
+   */
+  useExchangeRate(assetSymbol);
```

Comment:
Should this be moved into the provider or is there a reason it's here? A bit hard to tell what side effect it's having